lib/helpers.js

```diff
@@ -101,3 +99,43 @@ export function getMultibaseMultikeyHeader({value}) {
   }
   return value.slice(0, 4);
 }
+
+export function addX25519Context({contexts, keyAgreementKeyPair}) {
+  const {type} = keyAgreementKeyPair;
+  switch(type) {
+    case 'X25519KeyAgreementKey2019': {
+      if(!contexts.includes(X25519KeyAgreementKey2019.SUITE_CONTEXT)) {
+        contexts.push(X25519KeyAgreementKey2019.SUITE_CONTEXT);
+      }
+      break;
+    }
+    case 'X25519KeyAgreementKey2020': {
+      if(!contexts.includes(X25519KeyAgreementKey2020.SUITE_CONTEXT)) {
+        contexts.push(X25519KeyAgreementKey2020.SUITE_CONTEXT);
+      }
+      break;
+    }
+    default: {
+      throw new Error(`Unsupported key agreement key type, "${type}".`);
+    }
+  }
+}
+
+export async function getKeyPair({
+  fromMultibase, publicKeyMultibase, publicKeyDescription
+} = {}) {
+  let keyPair;
+  if(fromMultibase && publicKeyMultibase) {
+    keyPair = await fromMultibase({publicKeyMultibase});
+  } else {
+    keyPair = publicKeyDescription;
+  }
+  const {type} = keyPair;
+  let keyAgreementKeyPair;
+  if(type && (type === 'X25519KeyAgreementKey2020' ||
+    type === 'X25519KeyAgreementKey2019')) {
```

Comment:
Don't need the extra check to see if `type` is truthy.

```suggestion
  if(type === 'X25519KeyAgreementKey2020' ||
    type === 'X25519KeyAgreementKey2019') {
```

Reply:
I'll file an issue here ... since we don't want this to be X25519-specific.

Reply:
Updated conditions [65f3a59](https://github.com/digitalbazaar/did-method-key/pull/58/commits/65f3a59f975c1d101d14de3e2c4837a5dbffd263)